### PR TITLE
Feature/okta to auth0

### PR DESCRIPTION
**Pull request by kwnie**
# Description

- This is a transfer of login authorization services from Okta to Auth0
- User Story: [https://trello.com/c/zBSfWrsb](url)

## Fixes

- Removed all Okta files and libraries
- Added Auth0 libraries
- Rewrote middleware function to handle authorization and write new users to the database

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## Change Status

- Complete, but only tested in Insomnia and TablePlus (needs tests written)

### CODE REVIEW

**Comment by Antashma** [comment]
Looks good!  Great work removing the old okta files too!

**Review by da-vazquez — COMMENTED** [review]
Excellent Job Kelsey! Commits are organized and have clear descriptions. Swapping the middleware from Okta to Auth0 was not an easy task but you were able to make those modifications without a major overhaul to the existing code. Kudos

File: api/middleware/authRequired.js [hunk 1]
```diff
@@ -1,43 +1,45 @@
...
-const oktaJwtVerifier = new OktaJwtVerifier(oktaVerifierConfig.config);
 
-const makeParentObject = (claims) => {
+const makeParentObject = (user) => {
   return {
-    Email: claims.email,
-    Name: claims.name,
+    Name: user.name,
+    Email: user.email,
   };
```
Comment (SethMacP) [inline api/middleware/authRequired.js:13]: I am happy to see that the naming convention was updated to follow along with what we were using on the front end. We were all confused initially when we saw claims. Great catch.

File: api/middleware/authRequired.js [hunk 2]
```diff
@@ -1,43 +1,45 @@
...
+        audience: process.env.AUTH0_CLIENT_ID,
+        issuer: `https://${process.env.AUTH0_DOMAIN}/`,
+        algorithms: ['RS256'],
+      })
+    )(req, res, async () => {
+      const jwtUserObj = makeParentObject(req.user);
+      const parent = await Parents.findOrCreate(jwtUserObj);
+      if (parent) {
+        req.profile = parent;
+        next.apply(req, res);
```
Comment (SethMacP) [inline api/middleware/authRequired.js:38]: This was a fun one to see - I hadn't seen next.apply and it led to me also learning you can use next(parameter1, parameter2). Impressive work!

**Review by SethMacP — DISMISSED** [review]
Code looks clean & I saw you demo this live and saw no errors. Congrats - This was such a beast to figure out and I am so proud you persevered and completed it.

**Review by da-vazquez — DISMISSED** [review]

**Review by kdneldor — DISMISSED** [review]
Great job Kelsey! You did so much to pull the BE together for this transition to Auth0. Everything looks good. Well done!

**Comment by allraec** [comment]
Awesome job Kelsey! I've seen you work hard to get this transition figured out. Your clean and organized code is very easy to navigate around.

**Review by allraec — DISMISSED** [review]

**Review by Antashma — DISMISSED** [review]

**Comment by kwnie** [comment]
Rebased from main

**Review by kdneldor — APPROVED** [review]
Excellent PR Kelsey! Your commits are atomic and your changes are easy to understand and go through. Well done!

**Review by da-vazquez — APPROVED** [review]
Looks good! Conflicts have been resolved

**Review by Antashma — APPROVED** [review]
Looks good!  Thank you getting over that conflict with the okta config

**Pull request merged**